lib/laforge/activerecord.rb

```diff
@@ -105,6 +105,14 @@ def update_attribute_source_priority(attribute, source, priority)
       filter_loaded_data_entries(attributes: attribute, sources: source).each {|data_entry| data_entry.update(priority: priority) }
     end
 
+    def data_source?(source)
+      filter_loaded_data_entries(sources: source).any?
+    end
+
+    def data_source_names(attributes: nil)
+      filter_loaded_data_entries(attributes: attributes).map(&:source).map(&:name).uniq
```

Comment:
If there are tons of of data_entries will this be slow because we are loading the source for each entry? 

Maybe we should pluck/uniq the source_ids then only load the source once? 

Maybe rails already handles this and this is not a necessary optimization, I'm more asking for my own knowledge.